### ISSUE TITLE
Unexpected logic for the default store lookup URI

### ISSUE THREAD

**Issue by mattcaswell** [openssl/openssl#21068]
PR #18070 made changes to the default URI used for the store so that the default certificate directory is no longer used - so there is no default store added in this case.

This is arguably a breaking change since it also impacts cases where only the store lookup method is in use without specifying an alternative default URI and without the directory lookup method (in this case there will suddenly be no lookup method at all).

Strangely though specifying the SSL_CERT_DIR environment variable on the command line and specifying the same directory as would have been used anyway suddenly causes the default store URI to use that directory and the store lookup method does get added. The logic here seems unexpected.

**Comment by mattcaswell** [comment]
See also https://github.com/openssl/openssl/pull/20952#issuecomment-1564232273

**Comment by t8m** [comment]
Yeah, this is a breaking change in master. The default URI is winstore on Windows and nothing for other platforms. We need to discuss this within OTC if we want to revert it completely (on all platforms) or partially (everything except Windows).

**Comment by t8m** [comment]
These are the lines with the default store URI for reference:
https://github.com/openssl/openssl/blob/master/include/internal/common.h#L78-L82

**Comment by levitte** [comment]
I originally wrote a compliant URI parser for STORE use...  I wonder if I still have it around somewhere.

**Comment by t8m** [comment]
**OTC: We should revert the changes to the store handling code. We can keep the winstore provider but it would have to be loaded explicitly.**

It will be revisited post 3.2 release.
